braindecode/models/eegsym.py

```diff
@@ -0,0 +1,756 @@
+import mne
+from mne.channels import make_standard_montage
+from re import search
+
+from torch import nn
+from typing import List, Tuple
+from braindecode.models.base import EEGModuleMixin
+
+
+class EEGSym(EEGModuleMixin, nn.Module):
+    """EEGSym from Pérez-Velasco et al (2022) [eegsym2022]_.
+
+    .. figure:: https://raw.githubusercontent.com/Serpeve/EEGSym/refs/heads/main/EEGSym_scheme_online.png
+        :align: center
+        :alt: EEGSym Architecture
+
+    TO-DO: Use more EEGInceptionERP components.
+
+
+    Parameters
+    ----------
+    filters_per_branch : int, optional
+        Number of filters in each inception branch. Should be a multiple of 8.
+        Default is 8.
+    scales_time : tuple of int, optional
+        Temporal scales (in milliseconds) for the temporal convolutions in the first
+        inception module. Default is (500, 250, 125).
+    drop_prob : float, optional
+        Dropout probability. Default is 0.25.
+    activation : nn.Module, optional
+        Activation function to use. Default is nn.ELU().
+    spatial_resnet_repetitions : int, optional
+        Number of repetitions of the spatial analysis operations at each step.
+        Default is 1.
+    left_right_chs : list of tuple of str, optional
+        Optional list of tuples with the names of the left and right hemisphere channels.
+        If not provided, the channels will be split using function division_channels_idx,
+        and left/right channels will be matched by the function match_hemisphere_chans.
+    middle_chs : list of str, optional
+        Optional list of the names of the middle channels. If not provided, the channels
+        will be split using function division_channels_idx.
+
+    References
+    ----------
+    .. [eegsym2022] Pérez-Velasco, S., Santamaría-Vázquez, E., Martínez-Cagigal, V.,
+       Marcos-Martínez, D., & Hornero, R. (2022). EEGSym: Overcoming inter-subject
+       variability in motor imagery based BCIs with deep learning. IEEE Transactions
+       on Neural Systems and Rehabilitation Engineering, 30, 1766-1775.
+    .. [eegsym2022code] Pérez-Velasco, S., EEGSym source code.
+        https://github.com/Serpeve/EEGSym
+    """
+
+    def __init__(
+        self,
+        # braidecode parameters
+        n_chans=None,
+        n_outputs=None,
+        n_times=None,
+        chs_info=None,
+        input_window_seconds=None,
+        sfreq=None,
+        # Model parameters
+        filters_per_branch: int = 8,
+        scales_time: Tuple[int, int, int] = (500, 250, 125),
+        drop_prob: float = 0.25,
+        activation: nn.Module = nn.ELU(),
+        spatial_resnet_repetitions: int = 1,
+        left_right_chs: list[tuple[str, str]] | None = None,
+        middle_chs: list[str] | None = None,
+    ):
+        if (left_right_chs is None) != (middle_chs is None):
+            raise ValueError(
+                "Either both or none of left_right_chs and middle_chs must be provided."
+            )
```

Comment:
I should infer this based on the ch_info and assume some default values, but there is no default value on the model's parameter side. @Serpeve what are the 'core' middle channels that we can assume to be stacked?

Reply:
Yes if `left_right_chs` and `middle_chs` are not provided, they are automatically inferred based on the channel names. But it may happen that someone has data with non-standard channel names so I also left the option to provide them